library/src/main/java/com/okta/auth/OktaAuthenticationActivity.java

```diff
@@ -131,6 +129,9 @@ protected void onDestroy() {
         if (mTabManager != null) {
             mTabManager.dispose();
         }
+        if (resultIntent != null) {
+            LocalIntentBus.getInstance().post(OktaAuthManager.BROWSER_RESULT_CHANNEL, resultIntent);
```

Comment:
I don't think the resultIntent will be sent back to the starting activity if OktaAuthenticationActivity is destroyed and garbage collected. This can be due to memory pressure, configuration changes, or whenever the system wants to do gc. The resultIntent will not be posted to the starting activity since the starting activity will recreate a new instance of LocalIntentBus. Only way to prevent this is to have LocalIntentBus live in the Application or some other long lived object but this is not something we want to do.

Reply:
I don't this that this is a high risk possibility because of OktaAuthenticationActivity's short lifecycle after it receives Intent back from a browser or chrome tabs. Besides [LocalBroadcastReceiver](https://android.googlesource.com/platform/frameworks/support/+/android-support-lib-19.1.0/v4/java/android/support/v4/content/LocalBroadcastManager.java) follows the same strategy  Also, the scenario that you have described also going to kill anything within Application instance as within a Class instance.

Reply:
Meant the Application class not the context. The application will be a foreground process and won't be killed. LocalBroadcastManager is a different implementation, it is using a handler with the ui thread and posting to that message queue. Tried debugging with don't keep activities option on and it doesn't send the result back.

If the priority of keeping boiler plate code at a minimum outweigh these corner cases then we also can look at calling the result method directly instead.